rrmngmnt/host.py

```diff
@@ -335,14 +353,59 @@ def is_connective(self, tcp_timeout=20.0):
         :return: True if host is connective, False otherwise
         :rtype: bool
         """
-        ret = False
         try:
-            self.executor().run_cmd(['true'], tcp_timeout=tcp_timeout)
-            ret = True
-        except (socket.error, socket.timeout):
-            ret = False
-        except Exception:
-            self.logger.warning("Unexpected exception", exc_info=True)
-            ret = False
-
-        return ret
+            self.logger.info(
+                "Check if host %s is connective via ssh", self.fqdn
+            )
+            self.run_command(['true'], tcp_timeout=tcp_timeout)
+            return True
+        except (socket.timeout, socket.error) as e:
+            self.logger.debug("Socket error: %s", e)
+        except paramiko.SSHException as e:
+            self.logger.debug("SSH exception: %s", e)
+        self.logger.warning("Host %s does not connective via ssh", self.fqdn)
+        return False
+
+    def wait_for_ssh_connective_state(
+        self, positive, timeout=REBOOT_TIMEOUT, sample_time=REBOOT_SAMPLE_TIME
+    ):
+        """
+        Wait until host will be connective or not connective via ssh
+
+        :param positive: wait for the positive or negative connective state
+        :type positive: bool
+        :param timeout: wait timeout
+        :type timeout: int
+        :param sample_time: sample the ssh each sample_time seconds
+        :type sample_time: int
+        :return: True, if positive and ssh is connective or
+        negative and ssh does not connective, otherwise False
+        :rtype: bool
+        """
+        reachable = "unreachable" if positive else "reachable"
+        timeout_counter = 0
+        while self.is_connective() != positive:
+            if timeout_counter > timeout:
+                self.logger.error(
+                    "Host %s is still %s via ssh, after %s seconds",
+                    self.fqdn, reachable, timeout
+                )
+                return False
+            time.sleep(sample_time)
+            timeout_counter += sample_time
+        return True
+
+    def reboot(self):
+        """
+        Reboot host and wait until host will be connective via ssh
+        """
+        try:
+            self.run_command(["reboot"])
+        except socket.timeout as e:
+            self.logger.debug("Socket timeout: %s", e)
+        except paramiko.SSHException as e:
+            self.logger.debug("SSH exception: %s", e)
+        for positive in (False, True):
```

Comment:
I send it to function wait_for_ssh_connective_state, because first I wait until host will be not response via ssh and after until host will be response via ssh